src/module/rolls/chat-message.ts

```diff
@@ -68,47 +68,61 @@ export function formatRollMethod(rollMethod: RollMethod, stats: string[]) {
  * @param challengeDie1 The value of the first challenge die.
  * @param challengeDie2 The value of the second challenge die.
  */
-function computeRollOutcome(
+export function computeRollOutcome(
   score: number,
   challengeDie1: number,
   challengeDie2: number
 ): ROLL_OUTCOME {
-  if (score <= Math.min(challengeDie1, challengeDie2)) return ROLL_OUTCOME.MISS
-  if (score > Math.max(challengeDie1, challengeDie2))
-    return ROLL_OUTCOME.STRONG_HIT
-  return ROLL_OUTCOME.WEAK_HIT
+  return [challengeDie1, challengeDie2].filter(
+    (challengeDie) => score > challengeDie
+  ).length
+}
+
+/**
+ * Retrieves a localized string corresponding to a provided roll outcome.
+ * @param outcome The numeric outcome value to retrieve the string for.
+ * @param match Whether or not the outcome has matched challenge dice.
+ */
+export function computeOutcomeText(
+  outcome: ROLL_OUTCOME | DfRollOutcome,
+  match?: boolean | undefined
+) {
+  let outcomeKey = ROLL_OUTCOME[outcome]
+  if (match) {
+    outcomeKey += '_match'
+  }
+  return game.i18n.localize(`IRONSWORN.${outcomeKey}`)
 }
 
 /**
  * Will burning momentum improve the outcome of an Ironsworn action roll?
  * @param rawOutcome The outcome before burning momentum.
  * @param momentumOutcome The outcome after burning momentum.
  */
-function momentumBurnWouldUpgrade(
+export function momentumBurnWouldUpgrade(
   rawOutcome: ROLL_OUTCOME | undefined,
   momentumOutcome: ROLL_OUTCOME
 ) {
-  switch (true) {
-    case rawOutcome === ROLL_OUTCOME.MISS &&
-      momentumOutcome === ROLL_OUTCOME.WEAK_HIT:
-    case rawOutcome === ROLL_OUTCOME.MISS &&
-      momentumOutcome === ROLL_OUTCOME.STRONG_HIT:
-    case rawOutcome === ROLL_OUTCOME.WEAK_HIT &&
-      momentumOutcome === ROLL_OUTCOME.STRONG_HIT:
-      return true
-    default:
-      return false
-  }
+  return rawOutcome ? momentumOutcome > rawOutcome : false
 }
 
-type RollOutcomeKey = `${ROLL_OUTCOME}` | `${ROLL_OUTCOME}_match`
+type i18nOutcomeKey =
+  | `${keyof typeof ROLL_OUTCOME}`
+  | `${keyof typeof ROLL_OUTCOME}_match`
 
-function outcomeText(outcome: ROLL_OUTCOME, match: boolean) {
-  let key: RollOutcomeKey = outcome
+export function outcomeKey(
+  outcome: ROLL_OUTCOME,
+  match: boolean
+): i18nOutcomeKey {
+  let key: i18nOutcomeKey = ROLL_OUTCOME[outcome] as keyof typeof ROLL_OUTCOME
```

Comment:
Hmm. This is sneaky and clever, and it confused me for a bit. Is it a good idea to do `enum[1]`? Or is that well-defined behavior everywhere?

Reply:
hmm, good question. IME it works, but those have been different contexts. i can't recall reading anything either way as far as whether it's a good idea in all contexts. i'll look to see if i can determine what the consensus on the topic is.

Reply:
here's the TS documentation on reverse mappings in numeric enums, which includes the JS it generates for them. included is an example similar to `enum[1]`.

the JS is compiles to is basically a lookup function.

https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings

but the docs doing it makes a strong case, it doesnt necessarily make it a good idea, either. the opinion the author presents here is fairly partisan (in favour of TS enums), but they include links to several articles by people who hold the opposing view.

https://dev.to/dvddpl/whats-the-problem-with-typescript-enums-2okj

here's my take after digging in to those: this enum is  a concise way of representing a bidirectional lookup that's usable at runtime (unlike, say, a `type` expression). there's already several unidirectional lookup functions (and miscellaneous hash objects) being used for the same purpose in this system's code, so it's not really doing anything that's not already happening.

so, i think the takeaway here is: if you think the resulting compiled JS for this enum isn't especially offensive -- and has no marked disadvantages compared to a hand-coded lookup function -- i reckon we might as well use it :)